winthrop/books/migrations/0012_add_book_slug.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.13 on 2018-07-02 19:55
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+import django.db.models.deletion
+from django.utils.text import slugify
+from unidecode import unidecode
+
+
+def generate_book_slug(book):
+    '''Generate a slug based on first author, title, and year.
+
+    :rtype str: String in the format ``lastname-title-of-work-year``
+    '''
+    # get the first author, if there is one
+    author = book.creator_set.filter(creator_type__name='Author').first()
+    if author:
+        # use the last name of the first author
+        author = author.person.authorized_name.split(',')[0]
+    else:
+        # otherwise, set it to an empty string
+        author = ''
+    # truncate the title to first several words of the title
+    title = ' '.join(book.short_title.split()[:9])
```

Comment:
these still seem pretty long to me - do we think there's a danger of collision?

Reply:
Shortened to first 5 words in the title